main.go

```diff
@@ -154,7 +165,7 @@ func startPushServer(out chan<- *net.IP, localIp *net.IP) {
 
 	go func() {
 		err := s.ListenAndServe()
-		slog.Error("Server stopped", logging.ErrorAttr(err))
```

Comment:
I think the error logging worked better here, where the log can be more specific. Or maybe log in both places? Once for the error once for stopping from that error.

Reply:
My reason moving it was to allow for other things to use the context as well, the cause must be an `error` so it needs a log anyway if the context gets "done".

We could wrap the `ListenAndServe()` error and escalate it, if you want the additional information, like

```go
	go func() {
		err := s.ListenAndServe()
		cancel(errors.Join(errors.New("http server error"), err))
	}()
```

but I'm fine with any approach

Reply:
Yeah that could make sense, so the error source isn't lost.